pkg/operatorsource/configuring.go

```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 
 	marketplace "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	v1 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	client_marketplace "github.com/operator-framework/operator-marketplace/pkg/client"
```

Comment:
Why not `marketplace_client`? Adjectives generally go before nouns.

Reply:
Also, do we really need to import `"sigs.k8s.io/controller-runtime/pkg/client"` anymore? It seems like we just use it for the method signature. Can't we just put the `"github.com/operator-framework/operator-marketplace/pkg/client".Client` interface on the method signature, remove the other client import, and then remove this alias entirely? Then rather than having a thing called `clientMarketplace` at all, you could just call it `client` or `kubeClient` or whatever.

Reply:
@kevinrizza `NewConfiguringReconciler` takes a `client` of type `"sigs.k8s.io/controller-runtime/pkg/client".Client` as input, which is used to instantiate an object of the `"github.com/operator-framework/operator-marketplace/pkg/client".Client` interface. We need both imports here as the interface (`"github.com/operator-framework/operator-marketplace/pkg/client".Client`) is a wrapper for the `raw go-client` (`"sigs.k8s.io/controller-runtime/pkg/client".Client`)

Reply:
What's the point of using the interface if not to interpolate between types? If the client you wrote implements that interface, why not just update the method signature of `NewConfiguringReconciler` so that it uses the type instead of the interface? If the object that is passed to it also implements that interface it should compile.

Reply:
@kevinrizza the two interfaces are not the same. I named mine `Client` for convenience sake. It might help to think of the client I wrote as `MarketplaceClient`.

"github.com/operator-framework/operator-marketplace/pkg/client".Client  (let's call this `MarketplaceClient`):
```  
type Client interface {
	Create(ctx context.Context, obj runtime.Object) error
	Get(ctx context.Context, key ObjectKey, objExisting runtime.Object) error
	Update(ctx context.Context, obj runtime.Object) error
}
````
"sigs.k8s.io/controller-runtime/pkg/client".Client (Let's call this `RawClient`):

```
type Client interface {
	Reader
	Writer
	StatusClient
}
```

`MarketplaceClient` is a wrapper around `RawClient` as @tkashem suggested. They are two different interfaces, and if a function expects an input of type `RawClient` we cannot pass it an input of type `MarketplaceClient`. Hence we import both `RawClient` and `MarketplaceClient`. We get the `RawClient` from the factory, and internally pass it to `MarketplaceClient` such that configuring reconciler sees `MarketplaceClient` and not `RawClient`.